nikola/shortcodes.py

```diff
@@ -83,6 +80,9 @@ def _skip_whitespace(data, pos, must_be_nontrivial=False):
 
 def _skip_nonwhitespace(data, pos):
     """Return first position not before pos which contains a non-whitespace character."""
+    for i, x in enumerate(data[pos:]):
+        if x.isspace():
+            return pos + i
```

Comment:
Now you are iterating over the rest of the string twice in case no space is found.